packages/typegpu/src/smol/wgslGenerator.ts

```diff
@@ -428,6 +428,44 @@ ${alternate}`;
     }
   }
 
+  // 'j' stands for for (trust me)
+  if ('j' in statement) {
+    const [init, condition, update, body] = statement.j;
+    const initStatement = init ? generateStatement(ctx, init) : undefined;
+    const initStr = initStatement ? initStatement.slice(0, -1) : '';
+    const conditionExpr = condition
+      ? generateExpression(ctx, condition)
+      : undefined;
+    const conditionStr = conditionExpr ? resolveRes(ctx, conditionExpr) : '';
+    const updateStatement = update ? generateStatement(ctx, update) : undefined;
+    const updateStr = updateStatement ? updateStatement.slice(0, -1) : '';
+    const bodyStr = generateStatement(ctx, body);
+
+    return `\
+${ctx.pre}for (${initStr}; ${conditionStr}; ${updateStr})
+${bodyStr}
+${ctx.pre}`;
+  }
+
+  if ('w' in statement) {
+    const [condition, body] = statement.w;
+    const conditionStr = resolveRes(ctx, generateExpression(ctx, condition));
+    const bodyStr = generateStatement(ctx, body);
+
+    return `\
+      while (${conditionStr})
+        ${bodyStr}
+      `;
```

Comment:
Missing indent and dedent, as well as "pre"
```suggestion
${ctx.pre}while (${conditionStr})
${bodyStr}
${ctx.pre}`;
```